src/Serilog.Sinks.RavenDB/Sinks/RavenDB/RavenDBSink.cs

```diff
@@ -61,13 +80,35 @@ public RavenDBSink(IDocumentStore documentStore, int batchPostingLimit, TimeSpan
         /// <param name="events">The events to emit.</param>
         /// <remarks>Override either <see cref="PeriodicBatchingSink.EmitBatch"/> or <see cref="PeriodicBatchingSink.EmitBatchAsync"/>,
         /// not both.</remarks>
-        protected override async Task EmitBatchAsync(IEnumerable<Events.LogEvent> events)
+        protected override async Task EmitBatchAsync(IEnumerable<global::Serilog.Events.LogEvent> events)
         {
-            using (var session = _documentStore.OpenAsyncSession())
+            using (var session = string.IsNullOrWhiteSpace(_defaultDatabase) ? _documentStore.OpenAsyncSession() : _documentStore.OpenAsyncSession(_defaultDatabase))
             {
                 foreach (var logEvent in events)
                 {
-                    await session.StoreAsync(new LogEvent(logEvent, logEvent.RenderMessage(_formatProvider)));
+                    var logEventDoc = new LogEvent(logEvent, logEvent.RenderMessage(_formatProvider));
+                    await session.StoreAsync(logEventDoc);
+                    if (_expirationTimeSpan != null || _errorExpirationTimeSpan != null)
```

Comment:
The logic in the .ctor guarantees both of these will be true if either is, so may be less confusing just to check `_expiration` here?